### PR TITLE
Rest api

### PR DESCRIPTION
**Pull request by StephenDsouza90**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,21 +1,97 @@
...
 
-> what api provides to client
+A client uses an API to request for resources from the server. A server has the resources and transfers it to the client through an API. A resource can be any object for example, user, photo, number. An API allows one software to talk to another. It is a procedure allowing a client to access the data from a server. 
 
+An example of an API is "https://github.com/StephenDsouza90/demo-crud-sqlite-python-app" where:
+ 
+github.com/StephenDsouza90 - Endpoint URL
 
-##  The set up
+demo-crud-sqlite-python-app - Parameter
```
Comment (dsouzajude) [inline README.md]: `github.com` is the endpoint

`StephenDsouza90/demo-crud-sqlite-python-app` is the resource in which `/StephenDsouza90` is the parameter for the github username and `/demo-crud-sqlite-python-app` is the parameter for the github user's project name.

File: README.md [hunk 2]
```diff
@@ -1,21 +1,97 @@
...
+
+## Set up of rest.py 
+
+This app runs on a localhost (my computer) with an IP address of '0.0.0.0' (which maps to my computer) and a port 8080 as an endpoint.
+
+Example: 
+
+github.com - Localhost
+
+StephenDsouza90 - Endpoint
```
Comment (dsouzajude) [inline README.md]: This is wrong. Endpoint means IPAddress with a Port combination. Or DNS name (like google.com)  with a port. By default the port is 80 and it is never shown.

So google.com:80 would point to google.com and this is the endpoint.

The endpoint doesn't have and Paths or resourses. It's always either a DNS name or and IPAddress:Port combination like localhost:8080 is the endpoint.

Endpoint means a way to contact a server. A server can be contacted via the IP address or DNS Name and it's port usually by default 80. Where the IP Address or DNS name locates the server over the internet, it's like a server's ID and the port is thing on the server's machine where the server can listen to incoming requests from the client and process it.

File: README.md [hunk 3]
```diff
@@ -1,21 +1,97 @@
...
 ## How the app works
-> details on every function
-> its routing
-> how url path maps to the function
 
+In this app, the courses and students data has already been created in the courses_data and students_data objects respectively. The assignment_data has an empty list through which the client can alter it as required.
+
+### create_app()
+
+The create_app function has the following methods for
```
Comment (dsouzajude) [inline README.md]: So the `create_app()` function is also responsible for defining the routes and these routes would be mapped to the functions and these functions would then be responsible to process client requests and send back responses.

File: README.md [hunk 4]
```diff
@@ -1,21 +1,97 @@
...
+2. Get all assignments (GET request)
+3. Get courses of a particular student (GET request)
+4. Get total number of students in a particular course (GET request)
+5. Delete a particular student from a course (DELETE request)
+
+### routing
+
+The routing is showed below
+
+1. /courses for courses_data
```
Comment (dsouzajude) [inline README.md]: This is very abstract. You should say:

1. /courses for returning data about the courses saved in `courses_data` data structure.

File: README.md [hunk 5]
```diff
@@ -1,21 +1,97 @@
...
+2. /students for students_data
+3. /assignments for assignments_data
```
Comment (dsouzajude) [inline README.md]: This is very abstract. You should say:

1. /courses for returning data about the courses saved in `courses_data` data structure.

The same for 2. and 3.

File: README.md [hunk 6]
```diff
@@ -1,21 +1,97 @@
...
+### method type
+
+The method types are
+
+1. GET: -X GET
+2. POST: -X POST
+3. PUT: -X PUT
+4. DELETE: -XDELETE
+
+### url maping to function
```
Comment (dsouzajude) [inline README.md]: You should say the URL is mapped to the function via the `app.routes` decorator function which is part of the Flask Library. These functions are then responsible for processing client requests and returning responses.

File: README.md [hunk 7]
```diff
@@ -1,21 +1,97 @@
...
+4. Delete a particular student (DELETE request)
+5. Update a particular student (PUT request)
+
+assignments data:
+
+1. Assign a student to a course (POST request) 
+2. Get all assignments (GET request)
+3. Get courses of a particular student (GET request)
+4. Get total number of students in a particular course (GET request)
+5. Delete a particular student from a course (DELETE request)
```
Comment (dsouzajude) [inline README.md]: For each of these, give the whole curl command example.

Comment (StephenDsouza90) [reply README.md]: whole curl command given

File: README.md (hunk 1 above)
Comment (StephenDsouza90) [reply README.md]: change made to endpoint

File: README.md (hunk 2 above)
Comment (StephenDsouza90) [reply README.md]: change made to endpoint

File: README.md (hunk 6 above)
Comment (StephenDsouza90) [reply README.md]: mentioned app.route deco

File: README.md (hunk 5 above)
Comment (StephenDsouza90) [reply README.md]: change made

File: README.md (hunk 3 above)
Comment (StephenDsouza90) [reply README.md]: change made

**Review by dsouzajude — APPROVED** [review]

**Pull request merged**